drivers/led_strip/Kconfig.ws2812

```diff
@@ -7,93 +9,33 @@
 
 menuconfig WS2812_STRIP
 	bool "Enable WS2812 (and compatible) LED strip driver"
-	depends on SPI
+	select LED_STRIP_RGB_SCRATCH
 	help
 	  Enable LED strip driver for daisy chains of WS2812-ish
 	  (or WS2812B, WS2813, SK6812, or compatible) devices.
-	  These devices have a one-wire communications interface
-	  which encodes bits using pulses. Short pulses indicate
-	  zero bits, and long pulses indicate ones; refer to the
-	  chip datasheets for precise specifications. To implement
-	  this in a multitasking operating system, this driver
-	  generates the pulses using a SPI peripheral.
 
 if WS2812_STRIP
 
-config WS2812_STRIP_MAX_PIXELS
-	int "Maximum number of pixels in a strip"
-	default 12
-	help
-	  Set this to the maximum number of pixels you need
-	  to control at once. There is an 8x memory penalty associated
-	  with each increment of this value, so it's worth optimizing.
-
-config WS2812_STRIP_ONE_FRAME
-	hex "SPI frame to shift out to signal a one bit"
-	default 0x7c if SOC_SERIES_STM32F4X
-	default 0x70 if SOC_FAMILY_NRF
-	help
-	  When shifted out at the configured clock frequency,
-	  this must generate a pulse whose width fits within the chipset
-	  specifications for T1H, and whose interpulse timing meets low
-	  times. It is recommended that the first and last bits in the
-	  frame be zero; this "encourages" SPI IPs to leave MOSI low
-	  between frames.
-
-config WS2812_STRIP_ZERO_FRAME
-	hex "SPI frame to shift out to signal a zero bit"
-	default 0x60 if SOC_SERIES_STM32F4X
-	default 0x40 if SOC_FAMILY_NRF
-	help
-	  When shifted out at the configured clock frequency,
-	  this must generate a pulse whose width fits within the chipset
-	  specifications for T0H, and whose interpulse timing meets low
-	  times. It is recommended that the first and last bits in the
-	  frame be zero; this "encourages" SPI IPs to leave MOSI low
-	  between frames.
-
-# By default, we use GRBW [sic] (and ignore W).
-comment "The following options determine channel data order on the wire."
-
-config WS2812_RED_ORDER
-	int "Order in which a red pixel should be shifted out"
-	default 1
-	range 0 3
-	help
-	  If the red channel is shifted out first, specify 0.
-	  If second, specify 1, and so on.
+choice WS2812_STRIP_DRIVER
+prompt "Driver backend"
+default WS2812_STRIP_SPI
 
-config WS2812_GRN_ORDER
-	int "Order in which a green pixel should be shifted out"
-	default 0
-	range 0 3
-	help
-	  If the green channel is shifted out first, specify 0.
-	  If second, specify 1, and so on.
-
-config WS2812_BLU_ORDER
-	int "Order in which a blue pixel should be shifted out"
-	default 2
-	range 0 3
+config WS2812_STRIP_SPI
+	bool "Enable the SPI driver"
+	depends on SPI
```

Comment:
We could change it to `select`, there would be no need for `samples/drivers/led_ws2812/Kconfig` in this case?

Reply:
Since we don't have `HAS_SPI` using `select` here is not such a great idea.

Reply:
I think your second comment means I should disregard your first, which I will do. Please let me know if I'm wrong about that.